training/train.py

```diff
@@ -36,38 +36,46 @@
 lr = 0.03
 batch_size = 32
 
-
-def criteria_satisfied(_, current_epoch):
-    if current_epoch >= 10000:
-        return True
-    return False
-
-
 if overfit:
+    tag = "overfit"
     training_data = torch.utils.data.Subset(torch_dataset, range(10))
     lr = 0.05
     batch_size = 10
-
-    def criteria_satisfied(current_loss, _):
-        if current_loss < 1.0:
-            return True
-        return False
+    min_lr = 2e-3
+    patience = 100
+    stop_loss = 1.0
+    stop_epoch = None
+else:
+    tag = ""
+    min_lr = 1e-5
+    patience = 7
+    stop_loss = None
+    stop_epoch = 500
+
+
+def criteria_satisfied(current_loss, current_epoch):
+    if stop_loss is not None and current_loss < 1.0:
+        return True
+    if stop_epoch is not None and current_epoch > stop_epoch:
+        return True
+    return False
 
 
 device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
 model = ModelBuilder(alpha=0.25).to(device)
 
 parameters = list(model.parameters())
 optimizer = torch.optim.Adam(parameters, lr=lr)
+
 scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
     optimizer,
     mode="min",
-    factor=0.2,
-    patience=7,
+    factor=0.5,
+    patience=patience,
     threshold=1e-4,
     threshold_mode="rel",
     cooldown=1,
-    min_lr=1e-6,
+    min_lr=min_lr,
```

Comment:
I believe the purpose of the overfit mode is to determine the learning rate scheduler parameters that enable the fastest model training. Therefore, we will use the same parameters for training the model in non-overfit mode as well. However, I am not sure about it. @bulygin1985 please correct me.

Reply:
The purpose of overfit mode is to prove that your model design works - network is able to train, and give expected results in the end. 
There probably can be difference in learning rate for overfit and real training scenario, as overfit batch is 10 and real batches in training will something like 128 or 256 or whatever GPU RAM will allow.

Reply:
@kyrylogr description is closer to the truth. The overfit learning rate scheduler could differ from the training mode learning rate scheduler. The reason is different batch size between overfit and training modes. Roughly, multiplied batch size at "n" time leads to multiplied learning rate at "n" times. See: https://arxiv.org/pdf/1706.02677 , "2.1. Learning Rates for Large Minibatches"